### DISCUSSION TITLE
How to get the postmeta or menumeta in graphql?

### DISCUSSION THREAD

**Discussion by insaurabh** [wp-graphql/wp-graphql#2496] [GatsbyJS]
I am creating a menu on gatsby side. I called menus using  below snippet.

```
const data = useStaticQuery(graphql`{
allWpMenuItem(filter: {locations: {eq: PRIMARY}}, sort: {fields: order}) {
    nodes {
      id
      label
      title
      url
      path
      target
      parentId
    }
  }
}
`)

Now trying to get the custom meta fields attached to menus.

I know only below method by which i can add new fields to existing node

```
 add_action( 'graphql_register_types', function() {

    register_graphql_field( 'menuItem', 'color', [

       'type' => 'String',

       'description' => __( 'The color of the post', 'wp-graphql' ),

       'resolve' => function( $post ) {

        // print_r($post);  <---- I can see the menuItem object is dumped
        // print_r($post->ID); <--- nothing returned   

      
        return $image = get_post_meta($post->ID, '_menu_item_type', true );

       

      }

    ] );
    } );
 
 
 In the above snipped    $post->ID doesn't return anything.
 
 
 Any suggestion on this.
 
 Thanks.

**Comment by jasonbahl** [comment]
@insaurabh the `$post` is actually an instance of a WPGraphQL Model, so the database ID can be accessed using `$post->databaseId` instead of `$post->ID`.
